src/Workspaces/CoreTest/SemanticModelReuse/SemanticModelReuseTests.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.SemanticModelReuse
+{
+    [UseExportProvider]
+    public class SemanticModelReuseTests
+    {
+        private static Document CreateDocument(string code, string language)
+        {
+            var solution = new AdhocWorkspace().CurrentSolution;
+            var projectId = ProjectId.CreateNewId();
+            var project = solution.AddProject(projectId, "Project", "Project.dll", language).GetProject(projectId);
+
+            return project.AddMetadataReference(TestReferences.NetFx.v4_0_30319.mscorlib)
+                          .AddDocument("Document", SourceText.From(code));
+        }
+
+        #region C# tests
+
+        [Fact]
+        public async Task NullBodyReturnsNormalSemanticModel1_CSharp()
+        {
+            var document = CreateDocument("", LanguageNames.CSharp);
+
+            // trying to get a model for null should return a non-speculative model
+            var model = await document.ReuseExistingSpeculativeModelAsync(null, CancellationToken.None);
```

Comment:
Nit: named parameter. (Or add @akhera99's feature to GitHub, whichever is easier.)